### ISSUE TITLE
Should signin view redirect to / if already logged in

### ISSUE THREAD

**Issue by aristippe** [bread-and-pepper/django-userena#516]
Currently, it is possible to go to the login page even if a user is already logged in. Should it redirect if someone goes it while already logged in?

**Comment by pons1991** [comment]
No, It wont go. No user can visit the login page after the successful login.

**Comment by aristippe** [comment]
Then maybe I have some incorrect implementation since I can still the login page:

…/accounts/signin/
…/accounts/signin/?next=/ […]

**Comment by swistakm** [comment]
So yeah this is funny issue. In my opinion no one can say for sure what is the best approach for this case. For some it may be surprising behaviour but for others it may be a desirable feature in some projects:
> Tired of signing out your wive's session just to log into your account? Now you can just put your credentials while she is logged in and forget about "sign out" button!

But now let's be serious. In my personal opinion it's not a really big deal and depends on UX you are planning to provide. Probably you definitely should hide "sing in" button link for logged users if you don't want them to easily access this view. But what is the purpose in being so picky on limiting users freedom to access urls?

It is better to leave default behaviour that gives more flexibility than make opinionated decision. Also, this is already the default behaviour of `django.contrib.auth.login` view ([see code](https://github.com/django/django/blob/master/django/contrib/auth/views.py#L65)). It is easily configurable through settings passed in urlconf. Userena `signin` does not have such parameter but you can easily decorate it:

```python
from userena.views import signin

def my_signin_view(request, **kwargs):
    if request.user.is_authenticated():
        # do whatever you want
        return HttpResponseRedirect('whatever')
    else:
        return signin(request, **kwargs)
```

We can of course add this to `signin` view as an option but the default behaviour will most likely stay as is now for the following reasons:
* It is also the Django's `contrib.auth` default behaviour.
* Some developers may want being able to "sign in while being signed in" so such option must stay anyway
* What to do in that scenario is purely a matter of developer's/designer's preference so it is better to leave old behaviour as default than break backwards compatibility (IMHO)

If we consider adding such configurable option to `signin` view then we need to make few decisions:
* How to handle such attempts to view login page of signed users? Error? Redirect? Redirect where?
* Should we make it configurable through view params in url conf or maybe through custom userena settings variable?

I leave this discussion open because I would like to hear your opinion.
